lib/semian/protected_resource.rb

```diff
@@ -2,35 +2,59 @@ module Semian
   class ProtectedResource
     extend Forwardable
 
-    def_delegators :@resource, :destroy, :count, :semid, :tickets, :name
+    def_delegators :@bulkhead, :destroy, :count, :semid, :tickets, :name
     def_delegators :@circuit_breaker, :reset, :mark_failed, :mark_success, :request_allowed?,
                    :open?, :closed?, :half_open?
 
-    def initialize(resource, circuit_breaker)
-      @resource = resource
+    attr_reader :bulkhead, :circuit_breaker
+
+    def initialize(bulkhead, circuit_breaker)
+      @bulkhead = bulkhead
       @circuit_breaker = circuit_breaker
     end
 
     def destroy
-      @resource.destroy
-      @circuit_breaker.destroy
+      @bulkhead.destroy unless @bulkhead.nil?
+      @circuit_breaker.destroy unless @circuit_breaker.nil?
     end
 
     def acquire(timeout: nil, scope: nil, adapter: nil)
```

Comment:
not sure ttyt if we wanted to keep the logic together it would make sense but pollute the `acquire_*` methods with multiple yields - ex. below 

```
    def acquire(timeout: nil, scope: nil, adapter: nil)
      acquire_circuit_breaker(scope, adapter) do
        acquire_bulkhead(timeout, scope, adapter) do
          yield self
        end
      end
    end

    private

    def acquire_circuit_breaker(scope, adapter)
      yield self if @circuit_breaker.nil?

      @circuit_breaker.acquire do
        yield self
      end
    rescue ::Semian::OpenCircuitError
      Semian.notify(:circuit_open, self, scope, adapter)
      raise
    end

    def acquire_bulkhead(timeout, scope, adapter)
      yield self if @bulkhead.nil?

      @bulkhead.acquire(timeout: timeout) do
        Semian.notify(:success, self, scope, adapter)
        yield self
      end
    rescue ::Semian::TimeoutError
      Semian.notify(:busy, self, scope, adapter)
      raise
    end
```

thoughts?